src/sentry/incidents/logic.py

```diff
@@ -780,11 +785,12 @@ def delete_alert_rule(alert_rule):
         bulk_delete_snuba_subscriptions(list(alert_rule.query_subscriptions.all()))
         if incidents:
             alert_rule.update(status=AlertRuleStatus.SNAPSHOT.value)
-            for incident in incidents:
-                incident.update(status=IncidentStatus.CLOSED.value)
+            # Change the incident status asynchronously, which could take awhile with many incidents due to snapshot creations.
         else:
             alert_rule.delete()
 
+    tasks.auto_resolve_snapshot_incidents.apply_async(kwargs={"alert_rule_id": alert_rule.id})
```

Comment:
We should probably only do this if `alert_rule.id is not None`, since deleting a rule set the id to None.